specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2019-10-01/attestations.json

```diff
@@ -0,0 +1,779 @@
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Resource"
+        }
+      ],
+      "required": [
+        "properties"
+      ],
+      "description": "An attestation resource."
+    },
+    "AttestationProperties": {
+      "properties": {
+        "policyAssignmentId": {
+          "type": "string",
+          "description": "The resource ID of the policy assignment that the attestation is setting the state for."
+        },
+        "policyDefinitionReferenceId": {
+          "type": "string",
+          "description": "The policy definition reference ID from a policy set definition that the attestation is setting the state for."
+        },
+        "complianceState": {
+          "type": "string",
+          "enum": [
+            "Compliant",
+            "NonCompliant",
+            "Unknown"
+          ],
+          "x-ms-enum": {
+            "name": "ComplianceState",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Compliant",
+                "description": "The resource is in compliance with the policy."
+              },
+              {
+                "value": "NonCompliant",
+                "description": "The resource is not in compliance with the policy."
+              },
+              {
+                "value": "Unknown",
+                "description": "The compliance state of the resource is not known."
+              }
+            ]
+          },
+          "description": "The compliance state that should be set on the resource."
+        },
+        "expiresOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time the compliance state should expire."
+        },
+        "owner": {
+          "type": "string",
+          "description": "The person responsible for setting the state of the resource."
+        },
+        "comments": {
+          "type": "string",
+          "description": "Comments describing why this attestation was created."
+        },
+        "evidence": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AttestationEvidence"
+          },
+          "description": "The evidence supporting the compliance state set in this attestation."
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "The status of the attestation.",
+          "readOnly": true
+        },
+        "lastComplianceStateChangeAt": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The time the compliance state was last changed in this attestation.",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "policyAssignmentId"
```

Comment:
Why only policyAssignmentId is required? Don't we need complianceState and expiresOn? If they are not needed are we creating default values, Can we add it in comments.

Reply:
policyAssignmentId is required so the attestation is tied to a policy

For everything else there are no default values.  They are purely optional fields.  A user may want to create an attestation purely to assign an "owner" that is responsible for setting the state (and potentially expiresOn, etc...) on the attestation.  Or to leave a comment stating "I know this needs a state but we are still investigating what the state should be", etc...